### DISCUSSION TITLE
USB-C for Flexihub (MIDI and power supply, both for external MIDI controllers)

### DISCUSSION THREAD

**Discussion by ultracello** [Pirate-MIDI/Pirate-MIDI-Features-Bug-Tracking#63] [Feature Requests & Ideas]
More and more MIDI controllers will have both a power supply and MIDI via USB-C. A few days ago, I talked to several MIDI controller manufacturers at Superbooth 2022 in Berlin:

Electra One and Yaeltex are seems being interested and it looks like they will be using USB-C in the future, while Intech Design already uses USB-C for both MIDI and power on their MIDI controllers. The same goes for the Lehle Mono Volume S: MIDI and power supply run simultaneously via USB-C. Maybe Morningstar will switch later with upcoming models to USB-C too.

USB-C can generally handle up to 100W, if I'm not mistaken. So there is still some room for improvement, and manufacturers can finally install much brighter displays and illuminated buttons that can also be seen well during the day in sunlight.

In my opinion, it would be useful and somewhat future-proof for the next years to have at least 8x USB-C ports for connecting external MIDI controllers to the PirateMIDI Flexihub. It would be ideal to be able to power the Flexihub and all USB-C enabled MIDI controllers with just a single power bank (PD). 👍 Super minimalistic and you save adapters, cables, etc. (if you want to avoid WIDI for security reasons).

Yes, it's a controversial issue.

**Comment by simonaglover95** [comment]
USB type-C is only a connector shape. There are connectors with more pins, fewer pins suitable for different circuitry. USB Power Delivery is nothing really to do with the type-C connector - but internal power components and circuits in the device. 

All our current devices and future devices already use USB type-C and USB MIDI. 

USB MIDI is more complicated that you probably think because it is also linked to the function of USB. Not only MIDI. They are very very different to each other. If you have a device which can act as a USB host, I would suggest you just use a USB hub to connect your other USB MIDI devices. 

The Flexihub design is currently very compact. I would expect (but I am not the expert now - just guessing) that adding lots and lots of power circuitry for 8 different power delivery lanes would make such a device significantly larger. 

Also, would you suggest removing standard MIDI and Flexiports? Or are you suggesting a new alternative product to have as well as the "normal" Flexihub?
